File: backend/internal/domain/rating.go
package domain

import "context"

type Rating struct {
	ID        int64  `db:"id" json:"id"`
	ClientID  int64  `db:"client_id" json:"clientId"`
	MasterID  int64  `db:"master_id" json:"masterId"`
	SlotID    int64  `db:"slot_id" json:"slotId"`
	Score     int    `db:"score" json:"score"`
	CreatedAt string `db:"created_at" json:"createdAt"`
}

type RatingRepository interface {
	InsertRating(ctx context.Context, clientID, masterID, slotID int64, score int) (Rating, error)
	GetRatingByClientAndSlot(ctx context.Context, clientID, slotID int64) (Rating, error)
}

File: mobile/lib/core/models/rating.dart
class Rating {
  final int id;
  final int clientId;
  final int masterId;
  final int slotId;
  final int score;
  final String createdAt;

  const Rating({
    required this.id,
    required this.clientId,
    required this.masterId,
    required this.slotId,
    required this.score,
    required this.createdAt,
  });

  factory Rating.fromJson(Map<String, dynamic> json) => Rating(
    id: json['id'] as int,
    clientId: json['clientId'] as int,
    masterId: json['masterId'] as int,
    slotId: json['slotId'] as int,
    score: json['score'] as int,
    createdAt: json['createdAt'] as String,
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'clientId': clientId,
    'masterId': masterId,
    'slotId': slotId,
    'score': score,
    'createdAt': createdAt,
  };
}
